lib/utils/config.ts

```diff
@@ -13,77 +13,60 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import * as fs from 'fs';
-
 import Promise = require('bluebird');
 import ResinSdk = require('resin-sdk');
-import deviceConfig = require('resin-device-config');
 import * as semver from 'resin-semver';
 
 const resin = ResinSdk.fromSharedOptions();
 
-function readRootCa(): Promise<string | void> {
-	const caFile = process.env.NODE_EXTRA_CA_CERTS;
-	if (!caFile) {
-		return Promise.resolve();
-	}
-	return Promise.fromCallback(cb =>
-		fs.readFile(caFile, { encoding: 'utf8' }, cb),
-	)
-		.then(pem => Buffer.from(pem).toString('base64'))
-		.catch({ code: 'ENOENT' }, () => {});
-}
+type ImgConfig = {
+	applicationName: string;
+	applicationId: number;
+	deviceType: string;
+	userId: number;
+	username: string;
+	appUpdatePollInterval: number;
+	listenPort: number;
+	vpnPort: number;
+	apiEndpoint: string;
+	vpnEndpoint: string;
+	registryEndpoint: string;
+	deltaEndpoint: string;
+	pubnubSubscribeKey: string;
+	pubnubPublishKey: string;
+	mixpanelToken: string;
+	wifiSsid?: string;
+	wifiKey?: string;
+
+	// props for older OS versions
+	connectivity?: string;
+	files?: {
+		[filepath: string]: string;
+	};
+
+	// device specific config props
+	deviceId?: number;
+	uuid?: string;
+	registered_at?: number;
+};
 
 export function generateBaseConfig(
 	application: ResinSdk.Application,
-	options: { version?: string; appUpdatePollInterval?: number },
+	options: { version: string; appUpdatePollInterval?: number },
 ) {
-	if (options.appUpdatePollInterval) {
-		options = {
-			...options,
-			appUpdatePollInterval: options.appUpdatePollInterval * 60 * 1000,
-		};
-	}
+	options = {
+		...options,
+		appUpdatePollInterval: (options.appUpdatePollInterval || 10) * 60 * 1000,
+	};
 
-	return Promise.props({
-		userId: resin.auth.getUserId(),
-		username: resin.auth.whoami(),
-		apiUrl: resin.settings.get('apiUrl'),
-		vpnUrl: resin.settings.get('vpnUrl'),
-		registryUrl: resin.settings.get('registryUrl'),
-		deltaUrl: resin.settings.get('deltaUrl'),
-		apiConfig: resin.models.config.getAll(),
-		rootCA: readRootCa().catch(() => {
-			console.warn('Could not read root CA');
-		}),
-	}).then(results => {
-		return deviceConfig.generate(
-			{
-				application,
-				user: {
-					id: results.userId,
-					username: results.username,
-				},
-				endpoints: {
-					api: results.apiUrl,
-					vpn: results.vpnUrl,
-					registry: results.registryUrl,
-					delta: results.deltaUrl,
-				},
-				pubnub: results.apiConfig.pubnub,
-				mixpanel: {
-					token: results.apiConfig.mixpanelToken,
-				},
-				balenaRootCA: results.rootCA,
-			},
-			options,
-		);
-	});
+	return resin.models.os.getConfig(application.app_name, options) as Promise<
```

Comment:
This changes how api keys get inserted, because `getConfig` already includes them, and then we do our own injection on top of that using the below methods.

Have you tested all the cases? Using `--device <existing UUID>` at least I end up with a config.json that includes both `apiKey` and `deviceApiKey`. Definitely doesn't do that on master, I'm not sure if that will cause problems.